### PR TITLE
chore(cli): initial release

### PR DESCRIPTION
**Pull request by ElijahKotyluk**
Possible Flags: 
  - `--typescript` or `--ts`
  - `--preset` (eg: `--preset=<preset_option>` || `--preset <preset_option>`)
     - `default`
     - `base`
     - `essentials`

 
To test:

### Setup

```
git clone git@github.com:vuetifyjs/create-vuetify.git && cd create-vuetify && yarn
```

### JavaScript

```
yarn build && yarn start --preset=<preset_option>
```

### TypeScript
```
yarn build && yarn start --preset=<preset_option> --typescript
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -21,16 +25,21 @@ async function run () {
     alias: {
       'typescript': ['ts'],
     },
-    boolean: true,
   })
 
-  type PromptQuestions = 'projectName' | 'canOverwrite' | 'useTypeScript' | 'useRouter'
+  if (argv.preset && !validPresets.includes(argv.preset)) {
+    throw new Error(`'${argv.preset}' is not a valid preset. Valid presets are: 'base', 'default', 'essentials'.`)
```
Comment (johnleider) [inline src/index.ts:31]:
```suggestion
    throw new Error(`'${argv.preset}' is not a valid preset. Valid presets are: 'default', 'base', 'essentials'.`)
```

File: src/index.ts [hunk 2]
```diff
@@ -61,36 +64,48 @@ async function run () {
...
+          type: context.useTypeScript ? null : 'toggle',
           message: 'Use TypeScript?',
           active: 'Yes',
           inactive: 'No',
           initial: false,
         },
+        {
+          name: 'usePackageManager',
+          type: 'select',
+          message: 'Would you like to install dependencies with yarn or npm?',
```
Comment (johnleider) [inline src/index.ts:84]:
```suggestion
          message: 'Would you like to install dependencies with yarn, npm, or pnpm?',
```

File: src/index.ts [hunk 3]
```diff
@@ -61,36 +64,48 @@ async function run () {
...
+          initial: 0,
+          choices: [
+            { title: 'npm', value: 'npm' },
+            { title: 'pnpm', value: 'pnpm' },
+            { title: 'yarn', value: 'yarn' },
+            { title: 'none', value: null },
```
Comment (johnleider) [inline src/index.ts:90]:
```suggestion
            { title: 'yarn', value: 'yarn' },
            { title: 'npm', value: 'npm' },
            { title: 'pnpm', value: 'pnpm' },
            { title: 'none', value: null },
```

File: src/utils/installDependencies.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { spawnSync } from 'child_process'
+
+function installDependencies(projectRoot: string, packageManager: 'npm' | 'yarn' | 'pnpm') {
+  const cmd = (
+    packageManager === 'npm' ?
+      'npm install' :
+      packageManager === 'yarn' ?
+        'yarn' :
+        'pnpm install'
```
Comment (johnleider) [inline src/utils/installDependencies.ts:9]: what is packageManage is null?

File: template/typescript/base/src/main.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/**
+ * main.js
```
Comment (johnleider) [inline template/typescript/base/src/main.ts:2]:
```suggestion
 * main.ts
```

File: template/typescript/base/src/plugins/index.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/**
+ * plugins/index.js
```
Comment (johnleider) [inline template/typescript/base/src/plugins/index.ts:2]:
```suggestion
 * plugins/index.ts
```

File: template/typescript/base/src/plugins/index.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+ *
+ * Automatically included in `./src/main.js`
```
Comment (johnleider) [inline template/typescript/base/src/plugins/index.ts:4]:
```suggestion
 * Automatically included in `./src/main.ts`
```

File: template/typescript/base/src/plugins/vuetify.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+/**
+ * plugins/vuetify.js
```
Comment (johnleider) [inline template/typescript/base/src/plugins/vuetify.ts:2]:
```suggestion
 * plugins/vuetify.ts
```

File: template/typescript/base/src/plugins/webfontloader.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+/**
+ * plugins/webfontloader.js
```
Comment (johnleider) [inline template/typescript/base/src/plugins/webfontloader.ts:2]:
```suggestion
 * plugins/webfontloader.ts
```

File: template/typescript/default/src/App.vue [hunk 10]
```diff
@@ -1,25 +1,11 @@
...
-  data () {
-    return {
-      //
-    }
-  },
-})
-</script>
\ No newline at end of file
+<script setup lang="ts">
+  import HelloWorld from '@/components/HelloWorld'
```
Comment (johnleider) [inline template/typescript/default/src/App.vue:10]:
```suggestion
  import HelloWorld from '@/components/HelloWorld.vue'
```

File: template/typescript/default/src/components/HelloWorld.vue [hunk 11]
```diff
@@ -1,154 +1,79 @@
...
+    </v-responsive>
   </v-container>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue'
-
-// Logo
-import logo from '../assets/logo.svg'
+<script setup lang="ts">
```
Comment (johnleider) [inline template/typescript/default/src/components/HelloWorld.vue:78]:
```suggestion
  //
```

File: template/typescript/default/src/main.ts [hunk 12]
```diff
@@ -1,7 +1,23 @@
-import { createApp } from 'vue'
+/**
+ * main.js
```
Comment (johnleider) [inline template/typescript/default/src/main.ts:2]:
```suggestion
 * main.ts
```

File: template/typescript/default/src/plugins/index.ts [hunk 13]
```diff
@@ -0,0 +1,12 @@
+/**
+ * plugins/index.js
```
Comment (johnleider) [inline template/typescript/default/src/plugins/index.ts:2]:
```suggestion
 * plugins/index.ts
```

File: template/typescript/default/src/plugins/index.ts [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+ *
+ * Automatically included in `./src/main.js`
```
Comment (johnleider) [inline template/typescript/default/src/plugins/index.ts:4]:
```suggestion
 * Automatically included in `./src/main.ts`
```

File: template/typescript/default/src/plugins/vuetify.ts [hunk 15]
```diff
@@ -1,10 +1,26 @@
+/**
+ * plugins/vuetify.js
```
Comment (johnleider) [inline template/typescript/default/src/plugins/vuetify.ts:2]:
```suggestion
 * plugins/vuetify.ts
```

File: template/typescript/default/src/plugins/webfontloader.ts [hunk 16]
```diff
@@ -0,0 +1,15 @@
+/**
+ * plugins/webfontloader.js
```
Comment (johnleider) [inline template/typescript/default/src/plugins/webfontloader.ts:2]:
```suggestion
 * plugins/webfontloader.ts
```

File: template/typescript/essentials/src/components/HelloWorld.vue [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+            Community
+          </v-btn>
+        </v-col>
+      </v-row>
+    </v-responsive>
+  </v-container>
+</template>
+
+<script lang="ts" setup>
+
```
Comment (johnleider) [inline template/typescript/essentials/src/components/HelloWorld.vue:78]:
```suggestion
  //
```

File: template/typescript/essentials/src/main.ts [hunk 18]
```diff
@@ -0,0 +1,27 @@
+/**
+ * main.js
```
Comment (johnleider) [inline template/typescript/essentials/src/main.ts:2]:
```suggestion
 * main.ts
```

File: template/typescript/essentials/src/plugins/index.ts [hunk 19]
```diff
@@ -0,0 +1,11 @@
+/**
+ * plugins/index.js
```
Comment (johnleider) [inline template/typescript/essentials/src/plugins/index.ts:2]:
```suggestion
 * plugins/index.ts
```

File: template/typescript/essentials/src/plugins/index.ts [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+ *
+ * Automatically included in `./src/main.js`
```
Comment (johnleider) [inline template/typescript/essentials/src/plugins/index.ts:4]:
```suggestion
 * Automatically included in `./src/main.ts`
```

File: template/typescript/essentials/src/plugins/vuetify.ts [hunk 21]
```diff
@@ -0,0 +1,26 @@
+/**
+ * plugins/vuetify.js
```
Comment (johnleider) [inline template/typescript/essentials/src/plugins/vuetify.ts:2]:
```suggestion
 * plugins/vuetify.ts
```

File: template/typescript/essentials/src/plugins/webfontloader.ts [hunk 22]
```diff
@@ -0,0 +1,15 @@
+/**
+ * plugins/webfontloader.js
```
Comment (johnleider) [inline template/typescript/essentials/src/plugins/webfontloader.ts:2]:
```suggestion
 * plugins/webfontloader.ts
```

**Review by johnleider — CHANGES_REQUESTED** [review]

File: template/javascript/base/README.md [hunk 23]
```diff
@@ -0,0 +1,24 @@
+# base
+
+## Project setup
+```
+yarn install
```
Comment (jacekkarczmarczyk) [inline template/javascript/base/README.md:5]: could be selected package manager here (`yarn` - not `yarn install`, or `npm install` or `pnpm install`)

File: template/javascript/base/README.md [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+yarn build
+```
+
+### Lints and fixes files
+```
+yarn lint
+```
+
+### Customize configuration
+See [Configuration Reference](https://cli.vuejs.org/config/).
```
Comment (jacekkarczmarczyk) [inline template/javascript/base/README.md:24]:
```suggestion
See [Configuration Reference](https://vitejs.dev/config/).
```

File: template/typescript/base/package.json [hunk 25]
```diff
@@ -0,0 +1,29 @@
+{
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "serve": "vite preview",
+    "build": "vite build",
+    "lint": "eslint . --fix --ignore-path .gitignore",
+    "dev": "vite"
```
Comment (jacekkarczmarczyk) [inline template/typescript/base/package.json:8]: using same commands as in default vite project could be IMHO better
```suggestion
    "dev": "vite",
    "build": "vue-tsc --noEmit && vite build",
    "preview": "vite preview",
    "lint": "eslint . --fix --ignore-path .gitignore"
```

File: template/typescript/base/package.json [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "devDependencies": {
+    "@vitejs/plugin-vue": "^3.0.3",
+    "@vue/eslint-config-typescript": "^11.0.0",
+    "eslint": "^8.22.0",
+    "eslint-plugin-vue": "^9.3.0",
+    "sass": "^1.55.0",
+    "typescript": "^4.0.0",
+    "vite": "^3.1.8",
+    "vite-plugin-vuetify": "^1.0.0-alpha.12"
```
Comment (jacekkarczmarczyk) [inline template/typescript/base/package.json:27]:
```suggestion
    "vite-plugin-vuetify": "^1.0.0-alpha.12",
    "vue-tsc": "^1.0.9"
```

File: src/utils/installDependencies.ts (hunk 4 above)
Comment (ElijahKotyluk) [reply src/utils/installDependencies.ts:9]: This function won't except an undefined or null packageManager, this function will only ever get called if there is a selection made to be passed to it.

File: template/typescript/base/vite.config.ts [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+      '.json',
+      '.jsx',
+      '.mjs',
+      '.ts',
+      '.tsx',
+      '.vue',
+    ],
+  },
+  server: {
+    port: 3000,
```
[the remark this answers is not in the document]
Comment (ElijahKotyluk) [reply template/typescript/base/vite.config.ts:37]: Most of the template stuff is from John, not sure why he uses 3000 though.

**Review by johnleider — APPROVED** [review]

**Pull request merged**